expose inv_wishart_cholesky

spinkney opened the pull request:
#### Submission Checklist

- [] Run unit tests
- Documentation
    - [] If a user-facing facing change was made, the documentation PR is here: 

Adds `inv_wishart_cholesky_lpdf` and `inv_wishart_cholesky_rng`.

## Release notes

Added `inv_wishart_cholesky_lpdf` and `inv_wishart_cholesky_rng` distribution functions.

## Copyright and Licensing

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the BSD 3-clause license (https://opensource.org/licenses/BSD-3-Clause)

WardBrian commented on src/middle/Stan_math_signatures.ml:
```diff
@@ -296,6 +296,7 @@ let distributions =
   ; ([Lpmf; Rng; Log], "hypergeometric", [DInt; DInt; DInt; DInt], SoA)
   ; (full_lpdf_depr, "inv_chi_square", [DVReal; DVReal], SoA)
   ; (full_lpdf_depr, "inv_gamma", [DVReal; DVReal; DVReal], SoA)
+  ; ([Lpdf], "wishart_cholesky", [DMatrix; DReal; DMatrix], SoA)
```
I think you made a copy/paste error here which is why tests failed

WardBrian commented:
I think you need to run `make test; dune promote` to update the test files after fixing that typo

spinkney commented:
I keep getting this merlin wants to change as well
<img width="463" alt="Screen Shot 2022-05-19 at 2 10 40 PM" src="https://user-images.githubusercontent.com/23704057/169370955-8b52aefc-c7ea-469f-8aa7-4b5c39dfb441.png">

WardBrian commented:
That should have been fixed by #1198, have you pulled `master` recently?

spinkney commented:
yea and everytime I run `make` I get that file change

WardBrian commented:
oh, I see, that's undoing a change from #1197 which snuck in. Yeah, we should update to the version it wants to change to

WardBrian commented:
Does this depend on a specific math PR? The most recent test failure was 

```
test/expressions/tests3_test.cpp:74640:45: error: no member named 'inv_wishart_cholesky_rng' in namespace 'stan::math'; did you mean 'wishart_cholesky_rng'?
```

spinkney commented:
Yes, I was trying to do the expression tests for https://github.com/stan-dev/math/pull/2713. @rok-cesnovar said the PR is good. I'm probably just confused on how to get these tests to work. It's not in stan-math yet so how can I run them?

WardBrian commented:
If you have Jenkins permissions you can run them against specific PRs by doing to https://jenkins.flatironinstitute.org/job/Stan/job/Stanc3/job/PR-1202/build. If you don't have an account there, we can have @serban-nicusor-toptal set one up

I'll run it against that PR now!

spinkney commented:
@WardBrian I do have an account but the pw is saved somewhere I don't have access to at the moment. I'll be back there Sunday so please run for me in the meantime

WardBrian commented:
Oh, and expression testing is currently only for the functions, not distributions, so I think that is why that command has failed?

spinkney commented:
If I switch branches to main and I run make, switch to the stan-math main branch, it works for 
```
python3 runTests.py test/expressions/ --only-function "wishart_cholesky_lpdf"
```

WardBrian commented:
I think you can do 
`python3 runTests.py test/expressions/ --only-functions " inv_wishart_cholesky_lpdf(matrix, real, matrix) => real"` to bypass needing stanc3 entirely, otherwise you need to copy the built stanc to the math tests folder I believe

spinkney commented:
What I did was

Went to stan-math and cd'd into the make directory. I added a file named `local` and inside that I added STANC3=/location/to/stanc3 

I tried what you posted and I'm getting 6 errors 
```
test/expressions/tests0_test.cpp:20:45: error: no member named 'inv_wishart_cholesky_lpdf' in namespace 'stan::math'
auto result8 = stan::math::eval(stan::math::inv_wishart_cholesky_lpdf(matrix0_expr3,real1,matrix2_expr4));
                                ~~~~~~~~~~~~^
test/expressions/tests0_test.cpp:21:45: error: no member named 'inv_wishart_cholesky_lpdf' in namespace 'stan::math'
auto result9 = stan::math::eval(stan::math::inv_wishart_cholesky_lpdf(matrix5,real6,matrix7));
                                ~~~~~~~~~~~~^
test/expressions/tests0_test.cpp:45:45: error: no member named 'inv_wishart_cholesky_lpdf' in namespace 'stan::math'
auto result8 = stan::math::eval(stan::math::inv_wishart_cholesky_lpdf(matrix0_expr3,real1,matrix2_expr4));
                                ~~~~~~~~~~~~^
test/expressions/tests0_test.cpp:46:45: error: no member named 'inv_wishart_cholesky_lpdf' in namespace 'stan::math'
auto result9 = stan::math::eval(stan::math::inv_wishart_cholesky_lpdf(matrix5,real6,matrix7));
                                ~~~~~~~~~~~~^
test/expressions/tests0_test.cpp:78:45: error: no member named 'inv_wishart_cholesky_lpdf' in namespace 'stan::math'
auto result8 = stan::math::eval(stan::math::inv_wishart_cholesky_lpdf(matrix0_expr3,real1,matrix2_expr4));
                                ~~~~~~~~~~~~^
test/expressions/tests0_test.cpp:79:45: error: no member named 'inv_wishart_cholesky_lpdf' in namespace 'stan::math'
auto result9 = stan::math::eval(stan::math::inv_wishart_cholesky_lpdf(matrix5,real6,matrix7));
                                ~~~~~~~~~~~~^
6 errors generated.
```

WardBrian commented:
Is that in your PR or in the develop branch? I can’t imagine why it wouldn’t be able to find the function

I’m re-running against the math PR now, let’s see if the issue shows up in the CI

spinkney commented:
I don't know what the issues are that I'm having. I'm on the pr. Anyway, this passed!

WardBrian commented:
Ping me once the relevant math PR is merged and the docs are reviewed and we can get this in

WardBrian reviewed (approved):
LGTM! Thanks!
